core/services/bulletprooftxmanager/bulletprooftxmanager.go

```diff
@@ -59,13 +63,32 @@ func SendEther(s *strpkg.Store, from, to gethCommon.Address, value assets.Eth) (
 	return etx, err
 }
 
-func newAttempt(s *strpkg.Store, etx models.EthTx, gasPrice *big.Int) (models.EthTxAttempt, error) {
+func newAttempt(s *strpkg.Store, etx models.EthTx, suggestedGasPrice *big.Int) (models.EthTxAttempt, error) {
 	attempt := models.EthTxAttempt{}
 	account, err := s.KeyStore.GetAccountByAddress(etx.FromAddress)
 	if err != nil {
 		return attempt, errors.Wrapf(err, "error getting account %s for transaction %v", etx.FromAddress.String(), etx.ID)
 	}
 
+	gasPrice := s.Config.EthGasPriceDefault()
+	if suggestedGasPrice != nil {
+		gasPrice = suggestedGasPrice
+	}
+
+	if s.Config.OptimismGasFees() {
+		// Optimism requires special handling, it assumes that clients always call EstimateGas
+		callMsg := ethereum.CallMsg{To: &etx.ToAddress, From: etx.FromAddress, Data: etx.EncodedPayload}
+		gasLimit, estimateGasErr := s.EthClient.EstimateGas(context.TODO(), callMsg)
```

Comment:
I worry about this context here. Should newAttempt take a context now? If not, perhaps we can at least use `eth.DefaultQueryCtx()`

Reply:
I would be interested to see what behaviour we get on Optimism if client enables gas bumping now.